src/services/authorService.js

```diff
@@ -0,0 +1,102 @@
+import { Sequelize } from "sequelize";
+import Author from "../models/Author.js";
+import BookAuthor from "../models/BookAuthor.js";
+
+
+class AuthorServiceClass {
+    async createAuthor({ name, biography, bornDate }) {
+        return await Author.create({
+            name: name,
+            biography: biography,
+            bornDate: bornDate
+        });
+    }
+
+    async updateAuthor(id, { name, biography, bornDate }, loaders) {
+        const author = await Author.findByPk(id);
+        if (!author) {
+            throw new Error('Author not found');
+        }
+
+        await Author.update({
+            name: name,
+            biography: biography,
+            bornDate: bornDate
+        }, {
+            where: {
+                id: id
+            }
+        });
+
+        loaders.authorLoader.clear(id);
+
+        return await Author.findByPk(id);
+    }
+
+    async deleteAuthor(id, loaders) {
+        const author = await Author.findByPk(id);
+        if (!author) {
+            throw new Error('Author not found');
+        }
+
+        await Author.destroy({
+            where: {
+                id: id
+            }
+        });
+
+        loaders.authorLoader.clear(id);
+
+        return author;
+    }
+
+    async getAuthors({ filter = {}, page = 1, limit = 10 }) {
+        const offset = (page - 1) * limit;
+
+        const where = {};
+        const bookFilter = {};
+        let findByBook = false;
+        if(filter.bookId) {
+            bookFilter.bookId = filter.bookId;
+            findByBook = true;
+        }
+
+        const authorIds = await BookAuthor.findAll({
+            where: bookFilter
+        });
+
+        console.log("Author Ids: ", JSON.stringify(authorIds));
+
+        if (filter.bornDate) {
+            where.bornDate = filter.bornDate;
+        }
+
+        let authors = await Author.findAll({
+            where,
+            limit,
+            offset,
+            order: [['name', 'ASC']],
+        });
+
+        if(findByBook) {
+            authors = authors.filter(author => {
+                return authorIds.find(data => data.AuthorId === author.id);
+            });
+        }
+
+        const totalAuthors = authors.length;
+        console.log("Total authors: ", totalAuthors);
+
```

Comment:
IMP: This is where we should use field resolver, think of a Node / Type called Author, it would have a field called book, which in turn will call a field resolver, where you would get authorIds, so that you can just filter and return it. 

Otherwise this code will become lot more complex as we add more things, as you already see the if conditions which are added today.